### PR TITLE
feat(base): outputing chosen base items to JSON

### PR DESCRIPTION
**Pull request by dd84ai**
outputting base items to JSON with key `shop`
which players are wishing to show

by default all items are private
with 
`/shop public #NumberInShop`, item can become public
`/shop private #NumberInShop`, in order to hide item back

Also added for output `money`
which is needed in order to decide if the base is capable to buy things it made public

### CODE REVIEW

File: Plugins/Public/base_plugin/PlayerCommands.cpp [hunk 1]
```diff
@@ -1810,6 +1810,29 @@ namespace PlayerCommands
...
 			PrintUserCmdText(client, L"ERR Commodity does not exist");
 		}
+		else if (cmd == L"public" || cmd == L"private")
+		{
+			int item = ToInt(GetParam(args, ' ', 2));
+
+			int curr_item = 1;
+			for (map<UINT, MARKET_ITEM>::iterator i = base->market_items.begin(); i != base->market_items.end(); ++i, curr_item++)
+			{
+				if (curr_item == item)
```
Comment (dsyalex) [inline Plugins/Public/base_plugin/PlayerCommands.cpp:1820]: can we do some trick like `map<UINT, MARKET_ITEM>::iterator i = base->market_items.begin() + item - 1` (with a bounds check) or something to achieve the same result as this loop?

**Review by dsyalex — DISMISSED** [review]

**Review by dsyalex — COMMENTED** [review]
Shouldn't you also be adding this to the list of commands you can't run with viewshop?

File: Plugins/Public/base_plugin/ExportData.cpp [hunk 2]
```diff
@@ -156,12 +156,40 @@ void ExportData::ToJSON()
...
+				shop.write("min_stock", i->second.min_stock);
+				shop.write("max_stock", i->second.max_stock);
+
+				const GoodInfo* gi = GoodList::find_by_id(i->first);
+				wstring name = HkGetWStringFromIDS(gi->iIDSName);
+				shop.write("item", wstos(HtmlEncode(name)).c_str());
+			}
+		}
+		shop.close();
+		for (list<BasePassword>::iterator it = base->passwords.begin(); it != base->passwords.end(); ++it)
```
[the remark this answers is not in the document]
Comment (dd84ai) [reply Plugins/Public/base_plugin/ExportData.cpp:176]: indeed

**Comment by dd84ai** [comment]
> Shouldn't you also be adding this to the list of commands you can't run with viewshop?

Uh, yeah!

File: Plugins/Public/base_plugin/ExportData.cpp [hunk 3]
```diff
@@ -156,12 +156,40 @@ void ExportData::ToJSON()
 		}
 		pwds.close();
 
+		minijson::object_writer shop = pw.nested_object("shop");
+		int curr_item = 1;
+		for (map<UINT, MARKET_ITEM>::iterator i = base->market_items.begin(); i != base->market_items.end(); ++i, curr_item++)
+		{
+			if (i->second.is_exporting)
```
[the remark this answers is not in the document]
Comment (dd84ai) [reply Plugins/Public/base_plugin/ExportData.cpp:163]: I think I am too fat to do that

File: Plugins/Public/base_plugin/ExportData.cpp [hunk 4]
```diff
@@ -156,12 +156,40 @@ void ExportData::ToJSON()
...
+			{
+				shop.write("quantity", i->second.quantity);
+				shop.write("price", i->second.price);
+				shop.write("min_stock", i->second.min_stock);
+				shop.write("max_stock", i->second.max_stock);
+
+				const GoodInfo* gi = GoodList::find_by_id(i->first);
+				wstring name = HkGetWStringFromIDS(gi->iIDSName);
```
[the remark this answers is not in the document]
Comment (dd84ai) [reply Plugins/Public/base_plugin/ExportData.cpp:171]: I could wrap it into try Try-Block if you wish ;b

File: Plugins/Public/base_plugin/PlayerCommands.cpp (hunk 1 above)
Comment (dd84ai) [reply Plugins/Public/base_plugin/PlayerCommands.cpp:1820]: I think so

Comment (dd84ai) [reply Plugins/Public/base_plugin/PlayerCommands.cpp]: Yeah I was able to do that

**Comment by dd84ai** [comment]
Made dummy check with trying to load resulting JSON through Python, it still works.
silly in game check performed as well

File: Plugins/Public/base_plugin/ExportData.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (dsyalex) [reply Plugins/Public/base_plugin/ExportData.cpp:171]: That might be prudent

File: Plugins/Public/base_plugin/ExportData.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (dsyalex) [reply Plugins/Public/base_plugin/ExportData.cpp:163]: ?

[the remark this answers is not in the document]
Comment (dd84ai) [reply Plugins/Public/base_plugin/ExportData.cpp]: Removed if

File: Plugins/Public/base_plugin/ExportData.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (dd84ai) [reply Plugins/Public/base_plugin/ExportData.cpp:171]: wrapped to try catch

File: Plugins/Public/base_plugin/Main.h [hunk 5]
```diff
@@ -49,7 +49,7 @@ struct ARCHTYPE_STRUCT
 
 struct MARKET_ITEM
 {
-	MARKET_ITEM() : quantity(0), price(1.0f), min_stock(100000), max_stock(100000) {}
+	MARKET_ITEM() : quantity(0), price(1.0f), min_stock(100000), max_stock(100000), is_exporting(false) {}
```
Comment (dd84ai) [inline Plugins/Public/base_plugin/Main.h:52]: should be deleted, not really needed after the proposes changes

File: Plugins/Public/base_plugin/Main.h [hunk 6]
```diff
@@ -62,6 +62,9 @@ struct MARKET_ITEM
 
 	// Stop buying if the base holds more than this number of items
 	uint max_stock;
+
+	// Exporting to JSON for API access
+	bool is_exporting;
```
Comment (dd84ai) [inline Plugins/Public/base_plugin/Main.h:67]: should be deleted too

File: Plugins/Public/base_plugin/PlayerBase.cpp [hunk 7]
```diff
@@ -249,6 +249,7 @@ void PlayerBase::Load()
 						mi.price = ini.get_value_float(2);
 						mi.min_stock = ini.get_value_int(3);
 						mi.max_stock = ini.get_value_int(4);
+						mi.is_exporting = bool(ini.get_value_int(5));
```
Comment (dd84ai) [inline Plugins/Public/base_plugin/PlayerBase.cpp:252]: delete

File: Plugins/Public/base_plugin/PlayerBase.cpp [hunk 8]
```diff
@@ -395,8 +396,8 @@ void PlayerBase::Save()
 		for (map<UINT, MARKET_ITEM>::iterator i = market_items.begin();
 			i != market_items.end(); ++i)
 		{
-			fprintf(file, "commodity = %u, %u, %f, %u, %u\n",
-				i->first, i->second.quantity, i->second.price, i->second.min_stock, i->second.max_stock);
+			fprintf(file, "commodity = %u, %u, %f, %u, %u, %u\n",
+				i->first, i->second.quantity, i->second.price, i->second.min_stock, i->second.max_stock, int(i->second.is_exporting));
```
Comment (dd84ai) [inline Plugins/Public/base_plugin/PlayerBase.cpp:400]: delete

File: Plugins/Public/base_plugin/PlayerCommands.cpp [hunk 9]
```diff
@@ -1713,9 +1713,9 @@ namespace PlayerCommands
 					continue;
 				}
 				wchar_t buf[1000];
-				_snwprintf(buf, sizeof(buf), L"<TEXT>  %02u:  %ux %s %0.0f credits stock: %u min %u max</TEXT><PARA/>",
+				_snwprintf(buf, sizeof(buf), L"<TEXT>  %02u:  %ux %s %0.0f credits stock: %u min %u max (%s)</TEXT><PARA/>",
 					globalItem, i->second.quantity, HtmlEncode(name).c_str(),
-					i->second.price, i->second.min_stock, i->second.max_stock);
+					i->second.price, i->second.min_stock, i->second.max_stock, i->second.is_exporting ? L"Public" : L"Private");
```
Comment (dd84ai) [inline Plugins/Public/base_plugin/PlayerCommands.cpp:1718]: to be deleted

File: Plugins/Public/base_plugin/ExportData.cpp [hunk 10]
```diff
@@ -156,12 +156,35 @@ void ExportData::ToJSON()
...
+		minijson::object_writer shop = pw.nested_object("shop");
+		int curr_item = 1;
+		for (map<UINT, MARKET_ITEM>::iterator i = base->market_items.begin(); i != base->market_items.end(); ++i, curr_item++)
+		{
+			try {
+				shop.write("quantity", i->second.quantity);
+				shop.write("price", i->second.price);
+				shop.write("min_stock", i->second.min_stock);
+				shop.write("max_stock", i->second.max_stock);
+				shop.write("exporting", i->second.is_exporting);
```
Comment (dsyalex) [inline Plugins/Public/base_plugin/ExportData.cpp:168]: can we rename 'exporrting' and `is_exporting` to something that makes more sense please

**Review by dsyalex — CHANGES_REQUESTED** [review]

File: Plugins/Public/base_plugin/ExportData.cpp (hunk 10 above)
Comment (dd84ai) [reply Plugins/Public/base_plugin/ExportData.cpp]: renamed to `is_public`

**Review by dsyalex — DISMISSED** [review]

**Comment by dsyalex** [comment]
Ready for merge once author gives the go-ahead

**Comment by dd84ai** [comment]
manual CI is running:
- git clone git@github.com:dd84ai/FLHook.git?ref=base_shop_items_to_json
- Running VSCode
- installed boost `https://sourceforge.net/projects/boost/files/boost-binaries/1.64.0/boost_1_64_0-msvc-14.1-32.exe/download` to `C:\Users\user\repos\FLHook\Dependencies\boost`
- ls `C:\Users\user\repos\FLHook\Dependencies\boost`
```
user@WIN-GHKN5G5H3L9 MINGW64 ~/repos/FLHook/Dependencies/boost (base_shop_items_to_json)
$ ls
boost/     boost-build.jam  bootstrap.log  index.htm   Jamroot           LICENSE_1_0.txt  status/
boost.css  boostcpp.jam     bootstrap.sh*  index.html  lib32-msvc-14.1/  more/            tools/
boost.png  bootstrap.bat    doc/           INSTALL     libs/             rst.css
```
- copied active Freelancer folder
- built new dlls with Visual Studio 2017 (v141)
- cp C:\Users\user\repos\FLHook\Binaries\bin-vc14\FLHook.dll to C:\apps\FreelancerDev\EXE\FLHook.dll
- rm C:\apps\FreelancerDev\EXE\flhook_plugins\*.dll
- cp C:\Users\user\repos\FLHook\Binaries\bin-vc14\flhook_plugins\*.dll C:\apps\FreelancerDev\EXE\flhook_plugins
- launched server
```
Welcome to FLHook Console (3.1.0-UNKNOWN plugin)
Plugin loaded: hookext (ahookext.dll)
Plugin loaded: alley (alley.dll)
Plugin loaded: autobuy (autobuy.dll)
Plugin loaded: balancemagic (balancemagic.dll)
Plugin loaded: base (base.dll)
Plugin loaded: cloak (cloak.dll)
Plugin loaded: commoditylimit (commoditylimit.dll)
Plugin loaded: conn (conn.dll)
Plugin loaded: dronebay (dronebay.dll)
Plugin loaded: events (event.dll)
Plugin loaded: exhelp (exhelp.dll)
Plugin loaded: getrep (getrep.dll)
Plugin loaded: jsonbuddy (JSONBuddy.dll)
Plugin loaded: marketfucker (MarketFucker.dll)
Plugin loaded: minecontrol (minecontrol.dll)
Plugin loaded: misc (MiscellaneousCommands.dll)
Plugin loaded: dock (mobiledock.dll)
Plugin loaded: npc (npc.dll)
Plugin loaded: playercntl (playercntl.dll)
Plugin loaded: pvecontroller (pvecontroller.dll)
Plugin loaded: tempban (tempban.dll)
Plugin loaded: template (template.dll)
Loading character name cache
Loaded 0 names
Saving character name cache
Saved 0 names
DEBUG: Rep of li05_05_base is pf_09_grp
DEBUG: Rep of lI16_02_base is pf_09_grp
DEBUG: Rep of li05_03_base is li_lsf_grp
DEBUG: Rep of lI16_01_base is li_n_grp
DEBUG: Rep of lI16_03_base is li_n_grp
DEBUG: Rep of Hi01_01_base is fc_neutral
DEBUG: Rep of Hi02_01_base is fc_neutral
DEBUG: Rep of br04_04_base is co_be_grp
DEBUG: Rep of li02_04_base is fc_neutral
DEBUG: Rep of iw01_02_base is fc_neutral
DEBUG: Rep of li15_02_base is fc_neutral
AUTOBUY: Loaded 137 ammo limit entries
AUTOBUY: Loaded 18 FLHook extra items
AUTOBUY: Loaded 20 stackable launchers
BALANCEMAGIC: Loaded 29 damage adjusts.
CLOAK: Loaded 12 cloaking devices
CLOAK: Loaded 5 cloak disruptors
CLOAK: Loaded 0 limited cloak setups
CL: Loaded 0 Limited Commodities
DRONEBAY: 0 bayarches loaded.
DRONEBAY: 0 dronearches loaded.
EVENT: Loaded 0 events
EVENT DEBUG: Loaded 0 event data
EVENT DEBUG: Loaded 0 event player data
EVENT DEBUG: Loaded 248 IDs
HELP MENUS: Loaded 19 help menus
Rep: Loaded 148 factions
JSONBuddy: Loaded 552 ships
JSONBuddy: Loaded 248 IDs
MARKETFUCKER: Ignoring Bastille Guard Station.
MARKETFUCKER: Loaded 1349 ships for 453 bases
MARKETFUCKER: Loaded 20 items for 604 bases
DockingModules: Loaded 0 equipment
DockingModules: Allowing ships below the cargo capacity of 275 to dock
Resetrep: Loaded 148 factions
HYPERJUMP NOTICE: Ship Whitelist is 1 (1=On, 0=Off)
HYPERJUMP NOTICE: System Whitelist is 1 (1=On, 0=Off)
HYPERJUMP NOTICE: Survey Plane Limit is 15000
Jumpdrive [8]
Beacon Matrix [1]
IP Bans [0]
ID Bans [0]
Playercntl: Loaded 54 tag rephacks
Notice: Loaded 574 locked ships
PVECONTROLLER: NPC bounties are enabled.
PVECONTROLLER: Loaded 0 NPC bounty group scale values.
PVECONTROLLER: Loaded 0 NPC bounty classes.
PVECONTROLLER: Loaded 0 NPC bounty ship overrides.
PVECONTROLLER: Loaded 0 NPC bounty fighter armor multipliers.
PVECONTROLLER: Loaded 0 NPC bounty system scale multipliers.
PVECONTROLLER: Loaded 0 ship class types.
PVECONTROLLER: Loaded 0 NPC bounty class difference multipliers.
PVECONTROLLER: Loaded 0 NPC bounty warzone scale multipliers.
PVECONTROLLER: NPC drops are enabled.
PVECONTROLLER: Loaded 0 NPC drops by class.
TEMPLATE: Loaded 0 TEMPLATE
socket(ascii): socket connection listening
socket(unicode): socket connection listening
socket(encrypted-ascii): socket connection listening
socket(encrypted-unicode): socket connection listening
Rep: Loaded 148 factions
base: AffList was loaded succesfully.
```
- CI is in progress... standby

**Comment by dd84ai** [comment]
New CI run:
Diff
- cp C:\Users\user\repos\FLHook\Source\Release\FLHook.dll C:\apps\FreelancerDev\EXE\FLHook.dll
- rm C:\apps\FreelancerDev\EXE\flhook_plugins\*.dll
- find .dll in C:\Users\user\repos\FLHook\Plugins\Public and cp to C:\apps\FreelancerDev\EXE\flhook_plugins

- CI is running... standby

**Comment by dd84ai** [comment]
git checkout .
erasing target platform changes
```
-    <WindowsTargetPlatformVersion>10.0.17763.0</WindowsTargetPlatformVersion>
+    <WindowsTargetPlatformVersion>10.0.18362.0</WindowsTargetPlatformVersion>^
```
- git checkout .

- updating forked master in progress

**Comment by dd84ai** [comment]
New Build run:
- build... Release Win32
Failed. not installed sdk of requested version
- retargeting to present version
-    <WindowsTargetPlatformVersion>10.0.20348.0</WindowsTargetPlatformVersion>
+    <WindowsTargetPlatformVersion>10.0.18362.0</WindowsTargetPlatformVersion>
- Build failure. cannot override file in one of Release folders
```
Severity	Code	Description	Project	File	Line	Suppression State
Warning	C4244	'*=': conversion from 'float' to 'int', possible loss of data	PvE Controller	C:\Users\user\repos\FLHook\Plugins\Public\pvecontroller\Main.cpp	663	
Warning	C4244	'*=': conversion from '_Ty2' to 'int', possible loss of data	PvE Controller	C:\Users\user\repos\FLHook\Plugins\Public\pvecontroller\Main.cpp	672	
Warning	C4244	'*=': conversion from '_Ty2' to 'int', possible loss of data	PvE Controller	C:\Users\user\repos\FLHook\Plugins\Public\pvecontroller\Main.cpp	687	
Error	MSB6006	"link.exe" exited with code 1114.	HookExtension Plugin	C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets	1096	
Error	LNK1114	cannot overwrite the original file 'C:\Users\user\repos\FLHook\Plugins\Public\hookext_plugin\Release\ahookext.lib'; error code 5	HookExtension Plugin	C:\Users\user\repos\FLHook\Plugins\Public\hookext_plugin\LINK	1	
Warning	C4018	'>': signed/unsigned mismatch	Base Plugin	C:\Users\user\repos\FLHook\Plugins\Public\base_plugin\PlayerCommands.cpp	1826	
```
🔴

**Comment by dd84ai** [comment]
New build run:
- relaunched IDE
- launching build
✅ build is succesful. (hehe why)

**Comment by dd84ai** [comment]
New CI test run:
- cp C:\Users\user\repos\FLHook\Source\Release\FLHook.dll C:\apps\FreelancerDev\EXE\FLHook.dll
- rm C:\apps\FreelancerDev\EXE\flhook_plugins*.dll
- find .dll in C:\Users\user\repos\FLHook\Plugins\Public and cp to C:\apps\FreelancerDev\EXE\flhook_plugins
- launching server
first two times crashes, then it starts

**Comment by dd84ai** [comment]
- refloged branch before update with new master stuff
- https://github.com/dd84ai/FLHook/tree/json_before_rebase
Running CI for it...

Sold gold and food to base
receiving
user@WIN-GHKN5G5H3L9 MINGW64 /c/stats
$ python test.py
{'timestamp': '20230618T222400', 'bases': {'TB1929': {'passwords': ['hastesucks'], 'shop': {'quantity': 123, 'price': 1, 'min_stock': 100000, 'max_stock': 100000, 'is_public': False, 'item': 'Gin'}, 'affiliation': 'No Affiliation', 'type': 'legacy', 'solar': 'legacy', 'loadout': 'legacy', 'level': 1, 'money': 999047, 'health': 4.8233, 'defensemode': 1}}}
and broken /base interface
🔴

**Comment by dd84ai** [comment]
adding `dsacesrv.dll` back to folder, fixed most of issues.
Except incorrect JSON is still created. messed up data of all shop items into single dictionary with overriding keys to each other

**Comment by dd84ai** [comment]
https://github.com/DiscoveryGC/FLHook/pull/160/commits/98479250ca12407cb78846aa8d799716db9e1344
Fixed bugs
image

```
user@WIN-GHKN5G5H3L9 MINGW64 /c/stats
$ python test.py
{'timestamp': '20230618T232200', 'bases': {'TB1929': {'passwords': ['hastesucks'], 'shop_items': [{'quantity': 60, 'price': 1, 'min_stock': 100000, 'max_stock': 100000, 'is_public': False, 'item': 'Silencer CM Flare'}, {'quantity': 100, 'price': 1, 'min_stock': 100000, 'max_stock': 100000, 'is_public': True, 'item': 'Gadolinium'}, {'quantity': 1, 'price': 1, 'min_stock': 100000, 'max_stock': 100000, 'is_public': False, 'item': 'Coalition Thruster'}, {'quantity': 1, 'price': 1, 'min_stock': 100000, 'max_stock': 100000, 'is_public': False, 'item': 'Adv. Flashpoint Turret'}, {'quantity': 123, 'price': 1, 'min_stock': 100000, 'max_stock': 100000, 'is_public': False, 'item': 'Food Rations'}, {'quantity': 429, 'price': 1, 'min_stock': 100000, 'max_stock': 100000, 'is_public': False, 'item': 'Gold'}, {'quantity': 120, 'price': 1, 'min_stock': 100000, 'max_stock': 100000, 'is_public': False, 'item': 'Nanobots'}, {'quantity': 100, 'price': 1, 'min_stock': 100000, 'max_stock': 100000, 'is_public': False, 'item': 'Aluminium'}, {'quantity': 2, 'price': 1, 'min_stock': 100000, 'max_stock': 100000, 'is_public': False, 'item': 'Light Mining Array'}, {'quantity': 123, 'price': 1, 'min_stock': 100000, 'max_stock': 100000, 'is_public': False, 'item': 'Gin'}], 'affiliation': 'No Affiliation', 'type': 'legacy', 'solar': 'legacy', 'loadout': 'legacy', 'level': 1, 'money': 998941, 'health': 4.8113, 'defensemode': 1}}}
```
State is succesfully changed to Public, saved for server restart (persistent)
and succesfully oouputed to json and python is able to parse it correctly

**Comment by dd84ai** [comment]
CI rerun for rebased branch.
- test to parse json passes
- test to set public item in shop passes
- test to show info not broken in /base, /shop and etc passes
CI passes tests ✅

**Review by dsyalex — APPROVED** [review]

File: Plugins/Public/base_plugin/ExportData.cpp [hunk 11]
```diff
@@ -156,12 +156,36 @@ void ExportData::ToJSON()
...
+		for (map<UINT, MARKET_ITEM>::iterator i = base->market_items.begin(); i != base->market_items.end(); ++i, curr_item++)
+		{
+			try {
+				minijson::object_writer item = shop.nested_object();
+				item.write("quantity", i->second.quantity);
+				item.write("price", i->second.price);
+				item.write("min_stock", i->second.min_stock);
+				item.write("max_stock", i->second.max_stock);
+				item.write("is_public", i->second.is_public);
+
```
[the remark this answers is not in the document]
Comment (dd84ai) [reply Plugins/Public/base_plugin/ExportData.cpp:170]: agreed.

File: Plugins/Public/base_plugin/ExportData.cpp [hunk 12]
```diff
@@ -169,8 +169,12 @@ void ExportData::ToJSON()
 				item.write("is_public", i->second.is_public);
 				
 				const GoodInfo* gi = GoodList::find_by_id(i->first);
+				
 				wstring name = HkGetWStringFromIDS(gi->iIDSName);
-				item.write("item", wstos(HtmlEncode(name)).c_str());
+				item.write("name", wstos(HtmlEncode(name)).c_str());
```
Comment (dsyalex) [inline Plugins/Public/base_plugin/ExportData.cpp:174]: has occurred to me that we should not be doing HtmlEncode on this

**Review by dsyalex — CHANGES_REQUESTED** [review]
what did you do to the two vcxproj files?

File: Plugins/Public/base_plugin/ExportData.cpp (hunk 12 above)
Comment (dd84ai) [reply Plugins/Public/base_plugin/ExportData.cpp]: indeed. works without it

**Comment by dd84ai** [comment]
Manual CI:
```
$ python test.py
{'timestamp': '20230619T164500', 'bases': {'TB1929': {'passwords': ['hastesucks'], 'shop_items': [{'quantity': 60, 'price': 1, 'min_stock': 100000, 'max_stock': 100000, 'is_public': False, 'name': 'Silencer CM Flare', 'name_id': 501587, 'id': 2211913230, 'nickname': 'ge_s_cm_04_ammo'}, {'quantity': 100, 'price': 1, 'min_stock': 100000, 'max_stock': 100000, 'is_public': True, 'name': 'Gadolinium', 'name_id': 505870, 'id': 2222120073, 'nickname': 'commodity_gadolinium'}, {'quantity': 1, 'price': 1, 'min_stock': 100000, 'max_stock': 100000, 'is_public': False, 'name': 'Coalition Thruster', 'name_id': 520454, 'id': 2394587014, 'nickname': 'dsy_thruster_coal'}, {'quantity': 1, 'price': 1, 'min_stock': 100000, 'max_stock': 100000, 'is_public': True, 'name': 'Adv. Flashpoint Turret', 'name_id': 263498, 'id': 2759792129, 'nickname': 'co_turret01_mark03'}, {'quantity': 123, 'price': 1, 'min_stock': 100000, 'max_stock': 100000, 'is_public': False, 'name': 'Food Rations', 'name_id': 261622, 'id': 2863847232, 'nickname': 'commodity_food'}, {'quantity': 429, 'price': 1, 'min_stock': 100000, 'max_stock': 100000, 'is_public': False, 'name': 'Gold', 'name_id': 261610, 'id': 2866206539, 'nickname': 'commodity_gold'}, {'quantity': 120, 'price': 1, 'min_stock': 100000, 'max_stock': 100000, 'is_public': False, 'name': 'Nanobots', 'name_id': 263746, 'id': 2911012559, 'nickname': 'ge_s_repair_01'}, {'quantity': 100, 'price': 1, 'min_stock': 100000, 'max_stock': 100000, 'is_public': False, 'name': 'Aluminium', 'name_id': 505800, 'id': 2958831363, 'nickname': 'commodity_aluminium'}, {'quantity': 2, 'price': 1, 'min_stock': 100000, 'max_stock': 100000, 'is_public': False, 'name': 'Light Mining Array', 'name_id': 500189, 'id': 3103125006, 'nickname': 'hf_mining_gun'}, {'quantity': 123, 'price': 1, 'min_stock': 100000, 'max_stock': 100000, 'is_public': False, 'name': 'Gin', 'name_id': 505874, 'id': 3207769547, 'nickname': 'commodity_gin'}], 'affiliation': 'No Affiliation', 'type': 'legacy', 'solar': 'legacy', 'loadout': 'legacy', 'level': 1, 'money': 998941, 'health': 4.6325, 'defensemode': 1, 'shieldstate': 0}}}
```
✅ still green

**Review by dsyalex — APPROVED** [review]

**Pull request merged**